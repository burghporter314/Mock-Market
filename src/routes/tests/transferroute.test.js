import React, { useState } from 'react';
import renderer from 'react-test-renderer';
import { useAuth0 } from '@auth0/auth0-react';
import TransferPage from '../Transfer';

jest.mock("@auth0/auth0-react");

beforeEach(() => {
    useAuth0.mockReturnValue({
        isAuthenticated: true,
        user: "testuser",
        logout: jest.fn(),
        loginWithRedirect: jest.fn()
    })
})

it('should match snapshot', () => {

    const component = renderer.create(
            <TransferPage/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})