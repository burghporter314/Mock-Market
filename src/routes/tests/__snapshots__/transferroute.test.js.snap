// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Array [
  <div
    className="jumbotron"
    style={
      Object {
        "margin": "30px",
      }
    }
  >
    <div
      className="container main-form"
    >
      <form>
        <h2>
          Transfer Funds
        </h2>
        <br />
        <div
          className="col-md-6 mb-3 input-group"
        >
          <span
            className="input-group-text"
            id="basic-addon2"
          >
            Amount
          </span>
          <button
            className="btn btn-outline-secondary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            $1
          </button>
          <button
            className="btn btn-outline-secondary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            $10
          </button>
          <button
            className="btn btn-outline-secondary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            $100
          </button>
          <button
            className="btn btn-outline-secondary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            $1000
          </button>
          <span
            className="input-group-text"
            id="total-stock-amount"
          >
            $
          </span>
          <input
            aria-describedby="basic-addon2"
            aria-label="Enter Amount"
            className="form-control"
            id="input-amount"
            onChange={[Function]}
            placeholder="Enter Amount"
          />
          <button
            className="btn btn-outline-success"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Deposit
          </button>
          <button
            className="btn btn-outline-danger"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Withdrawal
          </button>
        </div>
        <hr />
        <div
          className="col-md-6 mb-3 input-group"
        >
          <span
            className="input-group-text"
            id="total-stock-amount"
          >
            Account Value: $0
          </span>
          <span
            className="input-group-text"
            id="estimated-value-deposit"
          >
            After Deposit: $0
          </span>
          <span
            className="input-group-text"
            id="estimated-value-withdrawal"
          >
            After Withdrawal: $0
          </span>
        </div>
      </form>
    </div>
  </div>,
  <br />,
  <div
    className="toast-container position-absolute top-50 start-50 translate-middle p-3"
  />,
]
`;
